### PR TITLE
Feature/improve gps accuraty

### PR DESCRIPTION
**Pull request by Alvaro2112**
I implemented a new method of getting the location using the GPS that is more accurate and that will also work in offline mode. I also added tests for previously untested code.

### CODE REVIEW

File: app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java [hunk 1]
```diff
@@ -119,28 +126,64 @@ protected void onCreate(Bundle savedInstanceState) {
...
         });
+        if (locationMode != null)
+            initLocationManager(locationMode);
+
     }
 
+    public void initLocationManager(String locationMode) {
+        LocationManager locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
+
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
```
Comment (gauthierboeshertz) [inline app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java:140]: You can do a simple helper function to fix the code climate issues

**Review by gauthierboeshertz — COMMENTED** [review]
Ok

File: .idea/misc.xml [hunk 2]
```diff
@@ -46,7 +46,11 @@
       </value>
     </option>
   </component>
+<<<<<<< HEAD
```
Comment (Warhedgehog) [inline .idea/misc.xml:49]: Those merge annotation should be deleted. I'm though clueless as to why this compiles at all

**Review by gauthierboeshertz — APPROVED** [review]

**Review by Warhedgehog — CHANGES_REQUESTED** [review]
Looks good to me. Just my one comment

File: .idea/misc.xml (hunk 2 above)
Comment (Alvaro2112) [reply .idea/misc.xml]: Done

**Review by Warhedgehog — APPROVED** [review]

**Pull request merged**